camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -77,6 +78,50 @@ def validate_openshift_report(cluster, details, deployments):
     assert set(cluster_nodes) == set(seen_nodes), "The number of expected nodes diverged"
 
 
+def validate_node_metrics(cluster, details):
+    node_metrics = details["sources"][0]["facts"][-1]["node_metrics"]
```

Comment:
"First source" and "last set of facts" are obviously brittle. What if there are more than one source? What if order of facts sets is changed?

Would it make sense to change that into a loop and check these assertions are true for any source / all sources? Could we iterate over facts sets and decide somehow which one is interesting for us (possibly based on namespace)?